index.js

```diff
@@ -71,7 +71,15 @@ function redisStore(args) {
       }
 
       if (opts.parse) {
-        result = JSON.parse(result);
+
+        try {
+          // allow undefined only if allowed by isCacheableValue
+          if(! ( (result === undefined || result === "undefined") && typeof args.isCacheableValue === 'function' && args.isCacheableValue(result))) {
```

Comment:
I guess if result = `undefined` but `undefined` is not a cacheable value:

`! ((true) && false) === true` and you end up JSON.parsing undefined.


If value is not defined, you get `null` from redis: https://github.com/NodeRedis/node_redis/blob/master/test/commands/get.spec.js#L84

So null can be parsed, as well as "null", and you can't get undefined but only "undefined", and this can't be parsed.

What about:
```js
if (result === "undefined") {
  result = undefined;
} else {
  result = JSON.parse(result);
}
```

I think if result is not cacheable, it should not be stored and thus not be retrieved, so maybe we can skip this? What do you think about it?